src/backup_restore.cpp

```diff
@@ -128,10 +128,165 @@ void BackupRestore::backupAndRestoreIpzVpd(types::IPZVpdMap& io_srcVpdMap,
+                ", is not found in the source path: " + i_srcPath);
+            continue;
+        }
+
+        if (!l_isEmptyDstVpdReceived &&
+            io_dstVpdMap.find(l_dstRecordName) == io_dstVpdMap.end())
+        {
+            logging::logMessage(
+                "Record: " + l_dstRecordName +
+                ", is not found in the destination path: " + i_dstPath);
+            continue;
+        }
+
+        types::BinaryVector l_defaultBinaryValue;
+        if (l_aRecordKwInfo.contains("defaultValue") &&
+            l_aRecordKwInfo["defaultValue"].is_array())
+        {
+            l_defaultBinaryValue =
+                l_aRecordKwInfo["defaultValue"].get<types::BinaryVector>();
+        }
+        else
+        {
+            logging::logMessage("Couldn't fetch default value for record: " +
+                                l_srcRecordName + ", keyword: " + l_srcKeyword);
+            continue;
+        }
+
+        bool l_isPelRequired = l_aRecordKwInfo.value("isPelRequired", false);
+
+        types::BinaryVector l_srcBinaryValue;
+        std::string l_srcStrValue;
+        if (!l_isEmptySrcVpdReceived)
+        {
+            utils::getKwVal(io_srcVpdMap.at(l_srcRecordName), l_srcKeyword,
+                            l_srcStrValue);
+            l_srcBinaryValue = types::BinaryVector(l_srcStrValue.begin(),
+                                                   l_srcStrValue.end());
+        }
+        else
+        {
+            // Read keyword value from DBus
+            // TODO: Uncomment when readKeyword API implemenation is done,
+            // call the API with required parameters.
+            /* l_srcBinaryValue = utils::readKeyword(i_srcPath,
+            * std::tuple<types::Record, types::Keyword>(l_srcRecordName,
+            l_srcKeyword)); l_srcStrValue =
+            std::string(l_srcBinaryValue.begin(),
+            l_srcBinaryValue.end());
+            */
+        }
+
+        types::BinaryVector l_dstBinaryValue;
+        std::string l_dstStrValue;
+        if (!l_isEmptyDstVpdReceived)
+        {
+            utils::getKwVal(io_dstVpdMap.at(l_dstRecordName), l_dstKeyword,
+                            l_dstStrValue);
+            l_dstBinaryValue = types::BinaryVector(l_dstStrValue.begin(),
+                                                   l_dstStrValue.end());
+        }
+        else
+        {
+            // Read keyword value from DBus
+            // TODO: Uncomment when readKeyword API implemenation is done,
+            // call the API with required parameters.
+            /* l_dstBinaryValue = utils::readKeyword(i_dstPath,
+            * std::tuple<types::Record, types::Keyword>(l_dstRecordName,
+            l_dstKeyword)); l_dstStrValue =
+            std::string(l_dstBinaryValue.begin(),
+            l_dstBinaryValue.end());
+            */
+        }
+
+        if (l_srcBinaryValue != l_dstBinaryValue)
+        {
+            if (l_dstBinaryValue == l_defaultBinaryValue)
```

Comment:
What happens if there is not concept of default value. Need to discuss this.

Reply:
I have added todo comment, will handle this in the separate PR.
Issue: https://github.com/ibm-openbmc/openpower-vpd-parser/issues/277